gtests/src/unit/tests/test_cluster.cpp

```diff
@@ -668,3 +694,89 @@ TEST_F(ClusterUnitTest, InvalidSsl) {
+  Future::Ptr connect_future(Memory::allocate<Future>());
+  ClusterConnector::Ptr connector(Memory::allocate<ClusterConnector>(contact_points,
+                                                                     PROTOCOL_VERSION,
+                                                                     bind_callback(on_connection_reconnect, connect_future.get())));
+
+  Future::Ptr up_future(Memory::allocate<Future>());
+  Future::Ptr recover_future(Memory::allocate<Future>());
+  RecoverClusterListener::Ptr listener(
+        Memory::allocate<RecoverClusterListener>(close_future, up_future, recover_future));
+
+  ClusterSettings settings;
+  settings.load_balancing_policy.reset(Memory::allocate<DCAwarePolicy>("dc1", 1, false)); // Allow connection to a single remote host
+  settings.load_balancing_policies.clear();
+  settings.load_balancing_policies.push_back(settings.load_balancing_policy);
+  settings.reconnect_timeout_ms = 1; // Reconnect immediately
+  settings.control_connection_settings.connection_settings.connect_timeout_ms = 200; // Give enough time for the connection to complete
+
+  connector
+      ->with_settings(settings)
+      ->with_listener(listener.get())
+      ->connect(event_loop());
+
+  ASSERT_TRUE(connect_future->wait_for(WAIT_FOR_TIME));
+  EXPECT_FALSE(connect_future->error());
+
+  // Notify every host as down
+  connect_future->cluster()->notify_down(local_address);
+  connect_future->cluster()->notify_down(remote_address);
+
+  // Notify the remote host as up
+  connect_future->cluster()->notify_up(remote_address);
+
+  // Verify that the remote host was marked as up
+  ASSERT_TRUE(up_future->wait_for(WAIT_FOR_TIME));
+  EXPECT_EQ(remote_address, listener->address());
+
+  cluster.stop(1); // Stop local node to very that remote host is tried for reconnection.
+
+  ASSERT_TRUE(recover_future->wait_for(WAIT_FOR_TIME));
+
+  connect_future->cluster()->close();
+  ASSERT_TRUE(close_future->wait_for(WAIT_FOR_TIME));
+
+  ASSERT_EQ(listener->connected_hosts().size(), 2u);
+  EXPECT_EQ(listener->connected_hosts()[0]->address(), Address("127.0.0.1", PORT));
+  EXPECT_EQ(listener->connected_hosts()[1]->address(), Address("127.0.0.2", PORT)); // Connected to remote host.
+}
+
+TEST_F(ClusterUnitTest, InvalidDC) {
+  mockssandra::SimpleCluster cluster(simple());
+  ASSERT_EQ(cluster.start_all(), 0);
+
+  ContactPointList contact_points;
+  contact_points.push_back("127.0.0.1");
+
+  Future::Ptr connect_future(Memory::allocate<Future>());
+  ClusterConnector::Ptr connector(Memory::allocate<ClusterConnector>(contact_points,
+                                                                     PROTOCOL_VERSION,
+                                                                     bind_callback(on_connection_connected, connect_future.get())));
+
+  ClusterSettings settings;
+  settings.load_balancing_policy.reset(Memory::allocate<DCAwarePolicy>("dc3", 0, false)); // Invalid DC and no using remote hosts
```

Comment:
I know `dc3` is not valid, but `dc2` is also invalid. What do you think about changing this to just `invalid_dc`?

`no using` => `not using`